### PR TITLE
(#6) Dictionary Landing Page

### LINKED ISSUE #6: Story: Dictionary Landing Page (URL is /)
## As a user, I would like to have a default (or “landing”) page when I first arrive at the Drug Dictionary so that I can understand what the dictionary contains, and choose to either browse or search the dictionary.

> **ESTIMATE** Without tests: 5, With tests: 8

### Acceptance Criteria

```gherkin
Scenario: User is able to access a default (or “landing”) page that displays all of the expected Dictionary elements for the drug dictionary. 
  Given user is on the landing page for the Drug Dictionary (/ URL)
    And page displays left navigation 
    And page title is “NCI Drug Dictionary”
    And {} introductory text appears below the page title
    And search bar is displayed below introductory text
  Then “Starts with” and “Contains” options are displayed
    And “Starts with” is selected by default
  Then “Browse” option is displayed below search bar
    And browse option contains all alphabetic letters  
    And “#” appears at the end of the list 
    And each option appears as a link 
    And the page is showing the expand results for letter “A”
    And the URL does not include expand=A
```

```gherkin
Scenario: Landing page metadata
  Given the user is viewing the dictionary landing page
  Then the title tag should be "NCI Drug Dictionary - National Cancer Institute"
   And the page contains meta tags with the following properties
            | property | content                           |
            | og:title | NCI Drug Dictionary   |
            | og:url   | http://localhost:3000/ |
            | robots   | index                           |
        And there is a canonical link with the href "https://www.cancer.gov/"  
  
```

**Analytics**
```gherkin
  Scenario: Page Load Analytics fires when a user views an expand result page
    Given "dictionaryTitle" is set to "NCI Drug Dictionary"
       And "baseHost" is set to "http://localhost:3000"
       And "canonicalHost" is set to "https://www.cancer.gov"
       And "siteName" is set to "National Cancer Institute"
       And "channel" is set to "Publications"
       And "analyticsPublishedDate" is set to "02/02/2011"
       And "analyticsName" is set to "Drug"
    When the user is viewing a results page based on clicking a letter like "M" in the dictionary
    Then search results page displays results title "# results found for: M"
    And browser waits
    Then there should be an analytics event with the following details
      | key                                    | value                                                      |
      | type                                   | PageLoad                                                   |
      | event                                  | DrugDictionaryApp:Load:ExpandResults                             |
      | page.name                              | www.cancer.gov/                                    |
      | page.title                             | NCI Drug Dictionary                                 |
      | page.metaTitle                         | NCI Drug Dictionary - National Cancer Institute |
      | page.language                          | english                                                    |
      | page.type                              | nciAppModulePage                                           |
      | page.channel                           | Publications                                               |
      | page.contentGroup                      | NCI Drug Dictionary                             |
      | page.publishedDate                     | 02/02/2011                                                 |
      | page.additionalDetails.dictionaryTitle | NCI Drug Dictionary                         |
      | page.additionalDetails.analyticsName   | Drug                                                |
      | page.additionalDetails.letter          | a                                                          |
      | page.additionalDetails.numberResults   | (int)Find it from the results                                                   |
```

### Resources:
* Web Pages:
    * https://www.cancer.gov/publications/dictionaries/cancer-drug

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- /expand has already been built

### Tasks
- [ ] Create basic search page with Search Box (no autosuggest), "Starts with" button, "Contains" button, and clickable A-Z list
- [ ] Display intro text on the dictionary page
- [ ] Set up default route to display expand=A (#10)
- [ ] Ensure page is set to indexable (robots = index) for search

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #6

* Added fetch call for char 'A' along with results and spinner, updated tracking event fire
* Added unit tests
* Tests for ticket/6

### CODE REVIEW

**Comment by seyilonge-nci** [comment]
@VictoriaSunNIH, @lburack, @blilianyu, @mworrest 
Please review https://react-app-dev.cancer.gov/drug-dictionary-app/pr-48/

**Comment by mworrest** [comment]
Looks good to me. Thanks!

**Comment by seyilonge-nci** [comment]
> Looks good to me from a styling perspective. Thanks!

Thanks @mworrest

**Review by karlikpj — APPROVED** [review]

File: cypress/integration/LandingPage.feature [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      | page.title                             | NCI Drug Dictionary                                 |
+      | page.metaTitle                         | NCI Drug Dictionary - National Cancer Institute |
+      | page.language                          | english                                                    |
+      | page.type                              | nciAppModulePage                                           |
+      | page.channel                           | Publications                                               |
+      | page.contentGroup                      | NCI Drug Dictionary                             |
+      | page.publishedDate                     | 02/02/2011                                                 |
+      | page.additionalDetails.dictionaryTitle | NCI Drug Dictionary                         |
+      | page.additionalDetails.analyticsName   | Drug                                                |
+      | page.additionalDetails.letter          | A                                                          |
```
Comment (kate-mashkina) [inline cypress/integration/LandingPage.feature:56]: Is there a reason the letter is capitalized? AC has it as lower cased

**Review by kate-mashkina — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**